src/Microsoft.DotNet.Interactive/Parsing/SubmissionParser.cs

```diff
@@ -38,26 +38,26 @@ public SubmissionParser(KernelBase kernel)
 
         public string DefaultLanguage { get; internal set; }
 
-        public PolyglotSyntaxTree Parse(string code)
+        public PolyglotSyntaxTree Parse(string code, string defaultLanguage = null)
         {
             var sourceText = SourceText.From(code);
 
             var parser = new PolyglotSyntaxParser(
-                sourceText, 
-                DefaultLanguage, 
+                sourceText,
+                defaultLanguage ?? DefaultLanguage,
```

Comment:
Incoming language service requests might have a different language specified, so we can't assume the default.

Reply:
Can we remove the default property then, so this is less confusing?